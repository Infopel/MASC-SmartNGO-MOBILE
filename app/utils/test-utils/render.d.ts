export type RenderOptions = {
  wrapper?: React.ComponentType<any>
  createNodeMock?: (element: React.ReactElement) => any
}
export type AllTheProvidersProps = { children: React.ReactNode }
