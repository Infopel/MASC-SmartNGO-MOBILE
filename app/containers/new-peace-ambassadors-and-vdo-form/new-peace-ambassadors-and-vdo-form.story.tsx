import * as React from "react"
import { storiesOf } from "@storybook/react-native"
import { StoryScreen, Story, UseCase } from "../../../storybook/views"
import { color } from "../../theme"
import { NewPeaceAmbassadorsAndVdoForm } from "./new-peace-ambassadors-and-vdo-form.component"

storiesOf("NewPeaceAmbassadorsAndVdoForm", module)
  .addDecorator((fn) => <StoryScreen>{fn()}</StoryScreen>)
  .add("Style Presets", () => (
    <Story>
      <UseCase text="Primary" usage="The primary.">
        <NewPeaceAmbassadorsAndVdoForm style={{ backgroundColor: color.error }} />
      </UseCase>
    </Story>
  ))
