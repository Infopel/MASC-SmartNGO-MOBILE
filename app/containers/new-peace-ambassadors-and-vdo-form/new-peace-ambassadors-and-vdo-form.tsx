import { withDatabase } from "@nozbe/watermelondb/DatabaseProvider"
import withObservables from "@nozbe/with-observables"
import { Button } from "components/button/button"
import { FormControlDate } from "components/form-control-date/form-control-date"
import { FormControlInput } from "components/form-control-input/form-control-input"
import { FormControlSeletor } from "components/form-control-seletor/form-control-seletor"
import { Formik, FormikHelpers } from "formik"
import { PeaceAmbassadorsAndVdoRoute } from "navigators"
import { goBack } from "navigators/navigation-utilities"
import * as React from "react"
import { compose } from "recompose"
import { createInitiative, CreateInitiativeParams } from "storage/mutations/initiative.mutations"
import { findAllDistricts, findAllProvinces, ILocation } from "storage/queries/location"
import * as yup from 'yup'


export interface NewPeaceAmbassadorsAndVdoFormProps {
  type: PeaceAmbassadorsAndVdoRoute
  provinces: ILocation[]
  districts: ILocation[]
}

/**
* Describe your component here
*/
const Container = function (props: NewPeaceAmbassadorsAndVdoFormProps) {
  const { provinces, districts, type } = props
  const validationSchema = yup.object().shape({
    provinceId: yup.string().required(),
    districtId: yup.string().required(),
    zone: yup.string().required(),
    locality: yup.string().required(),
    name: yup.string().required(),
    dateOfConstitution: yup.date().required(),
    numberOfMembers: yup.number().positive().required(),
    numberOfActiveMembers: yup.number().positive().required(),
  })

  async function create(values: CreateInitiativeParams, helpers: FormikHelpers<Partial<CreateInitiativeParams>>) {
    console.log({ values })
    await createInitiative({ ...values, type })
    goBack()
  }

  return (
    <Formik {...{ initialValues: {}, validationSchema, onSubmit: create }}>
      {({ handleSubmit }) => <>
        <FormControlInput label='common.name' fieldName="name" testID="name" />
        <FormControlSeletor fieldName="provinceId" testID="province" label='common.province' placeholder={'common.select-an-item'} items={provinces} />
        <FormControlSeletor fieldName="projectId" testID="projectId" label='common.project' placeholder={'common.select-an-item'} items={[]} />
        <FormControlSeletor fieldName="districtId" testID="district" label='common.district' placeholder={'common.select-an-item'} items={districts} />
        <FormControlInput label='common.zone' fieldName="zone" testID="zone" />
        <FormControlInput label='common.locality' fieldName="locality" testID="locality" />
        <FormControlDate label='new-peace-ambassadors-and-vdo.date-of-constitution' testID="dateOfConstitution" fieldName='dateConstitution' />
        <FormControlInput fieldName="numberOfMembers" testID="numberOfMembers" label='new-peace-ambassadors-and-vdo.number-of-members' />
        <FormControlInput fieldName="numberOfActiveMembers" testID="numberOfActiveMembers" label='new-peace-ambassadors-and-vdo.number-of-active-members' />
        <Button tx="common.save" testID="save" onPress={handleSubmit.bind(this, undefined)}></Button>
      </>}
    </Formik>
  )
}


export const NewPeaceAmbassadorsAndVdoFormContainer = compose(
  withDatabase,
  withObservables(['database'], ({ database }) => ({
    provinces: findAllProvinces(database), districts: findAllDistricts(database)
  }))
)(Container)