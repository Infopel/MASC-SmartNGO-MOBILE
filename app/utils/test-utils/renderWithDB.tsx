jest.mock("storage/adapter.ts")
import DatabaseProvider from "@nozbe/watermelondb/DatabaseProvider"
import { render } from "@testing-library/react-native"
import { default as React } from "react"
import { database } from "storage/database"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProvidersWithDB({ children }: AllTheProvidersProps) {
    return (
        <DatabaseProvider database={database}>
            <ThemeProvider>{children}</ThemeProvider>
        </DatabaseProvider>
    )
}

export function renderWithDB<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
    return render(ui, { wrapper: AllTheProvidersWithDB, ...options })
}


export class DiagnosticError extends Error {
    name: string = "Diagnostic error"
  }
  