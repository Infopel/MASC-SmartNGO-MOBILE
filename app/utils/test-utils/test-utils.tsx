import { render } from "@testing-library/react-native"
import React from "react"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProviders({ children }: AllTheProvidersProps) {
  return (
    <ThemeProvider>{children}</ThemeProvider>
  )
}

function customRender<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

// re-export everything
export * from "@testing-library/react-native"
// override render method
export { customRender as render }

