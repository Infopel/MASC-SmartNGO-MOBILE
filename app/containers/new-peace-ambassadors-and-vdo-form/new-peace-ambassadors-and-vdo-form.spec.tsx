import * as React from "react"
import { renderWithDB } from "utils/test-utils/renderWithDB"
import { NewPeaceAmbassadorsAndVdoFormContainer } from "./new-peace-ambassadors-and-vdo-form"

describe("<NewPeaceAmbassadorsAndVdoFormContainer>",()=>{
 it("render", async function () {
    const component = <NewPeaceAmbassadorsAndVdoFormContainer ></NewPeaceAmbassadorsAndVdoFormContainer>
    
    expect(renderWithDB(component).toJSON()).toMatchSnapshot()
  })
})
  