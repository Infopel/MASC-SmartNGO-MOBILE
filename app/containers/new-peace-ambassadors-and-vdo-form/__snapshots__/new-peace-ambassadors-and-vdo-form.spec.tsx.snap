// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewPeaceAmbassadorsAndVdoFormContainer> render 1`] = `null`;
